src/test/java/org/minions/devfund/danielmontecinos/moviedatabase/MovieDatabaseTest.java

```diff
@@ -0,0 +1,168 @@
+
+    /**
+     * Tests addMovie method only adds new actors to actorList.
+     */
+    @Test
+    public void testAddMovieMethodOnlyAddNewActorsToActorList() {
+        movieDatabase.addMovie("Movie 1", new String[] {"Actor 1", "Actor 2", "Actor 1", "Actor 2"});
+
+        final int expectedSize = 2;
+        assertEquals(expectedSize, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Tests addMovie method doesn't add existing actors to actorList.
+     */
+    @Test
+    public void testAddMovieMethodDoesNotAddSameActorsToActorList() {
+        movieDatabase.addMovie("Movie 9", new String[]{"Actor 1", "Actor 2"});
+        movieDatabase.addMovie("Movie 7", new String[]{"Actor 1", "Actor 2"});
+
+        final int expectedSize = 2;
+        assertEquals(expectedSize, movieDatabase.getActorList().size());
+    }
+
+    /**
+     * Tests addRating method adds a rating value to movie instance.
+     */
+    @Test
+    public void testAddRatingMethodAddsMovieRating() {
+        final double actualRating = 98.0;
+        movieDatabase.addMovie("Movie 7", new String[] {"Actor 6"});
+        movieDatabase.addRating("Movie 7", actualRating);
+
+        final double expectedRating = 98.0;
+        assertEquals(expectedRating, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Tests that updateRating methods updates rating previously set.
+     */
+    @Test
+    public void testUpdateRatingMethodUpdatesMovieRatingSetsBeforeUseIt() {
+        movieDatabase.addMovie("Movie 10", new String[] {"Actor 10"});
+        final double actualRating = 85.0;
+        movieDatabase.addRating("Movie 10", actualRating);
+
+        final double updatedRating = 100.0;
+        movieDatabase.updateRating("Movie 10", updatedRating);
+
+        final double expectedRating = 100.0;
+        assertEquals(expectedRating, movieDatabase.getMovieList().get(0).getRating(), 0.0);
+    }
+
+    /**
+     * Tests getBestActor method returns the actor with highest rating.
+     */
+    @Test
+    public void testGetBestActorReturnsTheActorWithHighestRating() {
+        final double midRating = 70.0;
+        movieDatabase.addMovie("Movie 1", new String[] {"Actor 1", "Actor 7"});
+        movieDatabase.addRating("Movie 1", midRating);
+
+        final double worstRating = 20.0;
+        movieDatabase.addMovie("Movie 2", new String[] {"Actor 9"});
+        movieDatabase.addRating("Movie 2", worstRating);
+
+        final double bestRating = 100.0;
+        movieDatabase.addMovie("Movie 7", new String[] {"Actor 5"});
+        movieDatabase.addRating("Movie 7", bestRating);
+        assertEquals("Actor 5", movieDatabase.getBestActor());
+    }
+
+    /**
+     * Tests getBestMovie returns movie with highest rating.
+     */
+    @Test
+    public void testGetsBesMovieReturnsTheMovieWithHighestRating() {
+        final double bestRating = 100.0;
+        movieDatabase.addMovie("Best Movie", new String[] {"Actor 1", "Actor 2", "Actor 21"});
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Best Movie" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)